test/reporters/tests/console.test.ts

```diff
@@ -66,3 +67,32 @@ stderr | console.test.ts
 global stderr afterAll`,
   )
 })
+
+test('interleave', async () => {
+  const logs: UserConsoleLog[] = []
+  const { stderr } = await runVitest({
+    root: './fixtures',
+    reporters: [
+      {
+        onUserConsoleLog(log) {
+          logs.push(log)
+        },
+      } satisfies Reporter,
+    ],
+  }, [resolve('./fixtures/console-interleave.test.ts')])
+  expect(stderr).toBe('')
+  expect(logs).toMatchObject([
+    {
+      type: 'stdout',
+      content: expect.stringContaining('1'),
+    },
+    {
+      type: 'stderr',
+      content: expect.stringContaining('2'),
+    },
+    {
+      type: 'stdout',
+      content: expect.stringContaining('3'),
+    },
+  ])
+})
```

Comment:
Could we run this with `forks` and `threads` pools using `test.each`? I think I've seen difference between these two pools in this empty line issue before. 🤔